modules/module-mongodb-storage/src/storage/implementation/MongoBucketBatch.ts

```diff
@@ -654,6 +655,10 @@ export class MongoBucketBatch
       return false;
     }
 
+    if (!createEmptyCheckpoints && this.persisted_op == null) {
```

Comment:
The default for `createEmptyCheckpoints` is true (to match the previous behaviour), which should result in this not being reached in most circumstances.